dtls_debug.h

```diff
@@ -129,8 +129,8 @@ void dtls_dsrv_log_addr(log_t level, const char *name, const session_t *addr);
 #define dtls_notice(...) LOG_INF(__VA_ARGS__)
 #define dtls_info(...) LOG_INF(__VA_ARGS__)
 #define dtls_debug(...) LOG_DBG(__VA_ARGS__)
-#define dtls_debug_hexdump(name, buf, length) { LOG_DBG("%s (%zu bytes):", name, length); LOG_HEXDUMP_DBG(buf, length, name); }
-#define dtls_debug_dump(name, buf, length) { LOG_DBG("%s (%zu bytes):", name, length); LOG_HEXDUMP_DBG(buf, length, name); }
+#define dtls_debug_hexdump(name, buf, length) { LOG_DBG("%s (%zu bytes):", name, (size_t)(length)); LOG_HEXDUMP_DBG(buf, length, name); }
+#define dtls_debug_dump(name, buf, length) { LOG_DBG("%s (%zu bytes):", name, (size_t)(length)); LOG_HEXDUMP_DBG(buf, length, name); }
```

Comment:
name should be wrapped as (name) in 2 places per line, as well as buf and length wrapped in LOG_HEXDUMP()
````
#define dtls_debug_hexdump(name, buf, length) { LOG_DBG("%s (%zu bytes):", (name), (size_t)(length)); LOG_HEXDUMP_DBG((buf), (length), (name)); }
#define dtls_debug_dump(name, buf, length) { LOG_DBG("%s (%zu bytes):", (name), (size_t)(length)); LOG_HEXDUMP_DBG((buf), (length), (name)); }
````

Reply:
If we want to start with applying general rules about macros, that will change much more ;-).

Reply:
Agreed.  However, every time we have to change a (#define) line of code, lets correctly fix that line and only correct the other lines that need doing when they need to be updated, so minimizing this particular fix.

Reply:
I disagree.
The tiny scope if this PR is based on an user request (see #212).
Extending the scope to "general/common rules", which are neither agreed nor discussed before is usually the way the generate quite a lot frustration, because that makes it impossible to improve the project in small steps.

If general rules about the design of macros should be applied to this project, I consider a separate issue for discussion will help much more than to "piggyback" this to tiny PRs.

Reply:
OK - what works best for you.